.openshift-ci/sensorversions.py

```diff
@@ -0,0 +1,120 @@
+        rawtags = apiresponse.json()['tags']
+        tags.extend(__filterQuayTags(rawtags))
+        if not bool(rawtags):
+            pageNotEmpty = False
+        page+=1
+    return set(tags)
+
+def __cliOutputToTags(stdoutput):
+    separated = stdoutput.decode(encoding="utf-8").splitlines()
+    return separated
+
+def __transformTagsToNumbers(tags):
+    numTags = []
+    for t in tags:
+        numTags.append(__splitVersion(t))
+    return sorted(numTags)
+
+def __splitVersion(version):
+    digits = re.search(r"(\d+)\.(\d+)\.\D*(\d+)", version)
+    return int(digits.group(1)), int(digits.group(2)), int(digits.group(3))
+
+def __getLatestTags(current_version, tags, num_versions):
+    numericaltags = __transformTagsToNumbers(tags)
+    _,y,_ = __splitVersion(current_version)
+    latestversions = []
+    ycurr = y
+    for tags in numericaltags[::-1]:
+        if tags[1] < y-num_versions+1:
+            break
+        if tags[1] < ycurr:
+            ycurr = tags[1]
+        if tags[1] == ycurr:
+            latestversions.append(str(tags[0]) + "." + str(tags[1]) + "." + str(tags[2]))
+            ycurr-=1
+    return latestversions
+
+# TODO: grab current_image from os.environ["MAIN_IMAGE_TAG"] after manual testing is done
+# getLastSensorVersionsFromQuay gets the latest patches of the last num_versions major versions
+# querying quay API is slow, prefer using git tag API
+def getLastSensorVersionsFromQuayAPI(current_version, num_versions):
+    tags = __queryQuayForTags()
+    return __getLatestTags(current_version, tags, num_versions)
+
+# TODO: grab current_image from os.environ["MAIN_IMAGE_TAG"] after manual testing is done
+# getLastSensorVersionsFromGitTagsAPI gets the latest patches of the last num_versions major versions via Git API
+# much faster than querying quay
+def getLastSensorVersionsFromGitTagsAPI(current_version, num_versions):
+    apiresponse = requests.get("https://api.github.com/repos/stackrox/stackrox/git/refs/tags")
+    tags = __filterGitTags(apiresponse.json())
+    return __getLatestTags(current_version, tags, num_versions)
+
+# getLastSensorVersionsFromGitTagsCLI gets the latest patches of the last num_versions major versions via Git CLI
+# preferably use this to avoid API calls if possible
+def getLastSensorVersionsFromGitTagsCLI(current_version, num_versions):
+    rawtags = __cliOutputToTags(subprocess.check_output(["git", "tag", "--list"]))
+    tags = __filterGitCLITags(rawtags)
+    return __getLatestTags(current_version, tags, num_versions)
+
+def main(argv):
+    latestversions = getLastSensorVersionsFromGitTagsCLI(argv[1], 4)
+    #latestversions = getLastSensorVersionsFromGitTagsAPI(argv[1], 4)
+    #latestversions = getLastSensorVersionsFromQuayAPI(argv[1], 4)
+    printversions = ""
+    for version in latestversions:
+        printversions += str(version) + " "
+    print(printversions)
```

Comment:
This can be simply
```suggestion
print(" ".join(printversions))
```

However, I think that if you plan to consume the output of this CLI, it may be more convenient to separate versions with newlines.

Reply:
This was just for manual testing of the functions, I'll remove the main function and prints later